src/core/core.controller.js

```diff
@@ -457,7 +457,7 @@ module.exports = function(Chart) {
 
 			// Draw each dataset via its respective controller (reversed to support proper line stacking)
 			helpers.each(me.data.datasets, function(dataset, datasetIndex) {
-				if (me.isDatasetVisible(datasetIndex)) {
+				if (me.isDatasetVisible(datasetIndex) && me.isDatasetVisible(datasetIndex).hasOwnProperty('controller')) {
```

Comment:
`me.getDatasetMeta(datasetIndex).controller` **must** exists at this point, having a check here only workaround the real issue. Also, `me.isDatasetVisible(datasetIndex)` returns a boolean, so this condition is always false.

Can you create a fiddle that reproduce the issue?